xmas-lights-flask/app.py

```diff
@@ -0,0 +1,225 @@
+import math
+import random
+import time
+import json
+from pathlib import Path
+import threading
+import traceback
+
+from flask import Flask, request, abort
+
+import app_html
+
+
+max_leds = 40
+# led_strip = neopixel.NeoPixel(max_leds, 0, 22, 'GRB')  # TODO
+
+# Global scope for script interpretation
+GLOBAL_SCOPE = {
+    'math': math,
+    'random': random,
+    '__builtins__': {
+        'abs': abs,
+        'all': all,
+        'any': any,
+        'bin': bin,
+        'bool': bool,
+        'bytearray': bytearray,
+        'bytes': bytes,
+        'callable': callable,
+        'chr': chr,
+        'complex': complex,
+        'delattr': delattr,
+        'dict': dict,
+        'dir': dir,
+        'divmod': divmod,
+        'enumerate': enumerate,
+        'filter': filter,
+        'float': float,
+        'frozenset': frozenset,
+        'getattr': getattr,
+        'globals': globals,
+        'hasattr': hasattr,
+        'hash': hash,
+        'help': help,
+        'hex': hex,
+        'id': id,
+        'int': int,
+        'isinstance': isinstance,
+        'issubclass': issubclass,
+        'iter': iter,
+        'len': len,
+        'list': list,
+        'locals': locals,
+        'map': map,
+        'max': max,
+        'memoryview': memoryview,
+        'min': min,
+        'next': next,
+        'object': object,
+        'oct': oct,
+        'ord': ord,
+        'pow': pow,
+        'print': print,
+        'property': property,
+        'range': range,
+        'repr': repr,
+        'reversed': reversed,
+        'round': round,
+        'set': set,
+        'setattr': setattr,
+        'slice': slice,
+        'sorted': sorted,
+        'str': str,
+        'sum': sum,
+        'super': super,
+        'tuple': tuple,
+        'type': type,
+        'zip': zip,
+    }
+}
+
+
+reset = True
+
+patterns = {}
+
+try:
+    pattern_filename = Path(__file__).parent / "patterns.json"
```

Comment:
Added this to make it relative to the app.py file, rather than depending on what my working dir is